### ISSUE TITLE
fails to load bson, fails to load fixtures using pure js

### ISSUE THREAD

**Issue by jskye** [DemocracyOS/democracyos#1307]
Running master on mac.

`make run`
```
{ Error: Cannot find module '../build/Release/bson'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/juliusskye/votebird/node_modules/bson/ext/index.js:15:10)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3) code: 'MODULE_NOT_FOUND' }
js-bson: Failed to load c++ bson extension, using pure JS version
```

It loads fine using the pure js version.
But then i get another error when attempting to load fixtures,

`node ./bin/dos-db load tag ./lib/fixtures/tags.json`
```
{ Error: Cannot find module '../build/Release/bson'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/juliusskye/votebird/node_modules/bson/ext/index.js:15:10)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3) code: 'MODULE_NOT_FOUND' }
js-bson: Failed to load c++ bson extension, using pure JS version
Connecting to mongodb://localhost:27017/DemocracyOS-dev
Register mongoose models.
/Users/juliusskye/votebird/node_modules/mongoose/node_modules/mongodb/lib/mongodb/db.js:299
          throw err;
          ^

Error: Cannot find module 'lib/utils'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/juliusskye/votebird/lib/db-api/tag.js:8:13)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/juliusskye/votebird/lib/db-api/proposal.js:7:14)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
```

**Comment by jskye** [comment]
seems to be similar to this error[https://github.com/Automattic/mongoose/issues/2906](url)
Not sure how to fix though.

**Comment by jskye** [comment]
tried most of these solutions and havent found one thats worked yet:
http://stackoverflow.com/questions/21656420/failed-to-load-c-bson-extension/22242472#22242472
Get the same problem on latest release branch

**Comment by jskye** [comment]
Actually i fixed the bson module problem with this hack (from above link):

> When you install the mongoose module by npm, it does not have a built bson module in it's folder. In the file node_modules/mongoose/node_modules/mongodb/node_modules/bson/ext/index.js, change the line
> 
> bson = require('../build/Release/bson');
> to
> 
> bson = require('bson');
> and then install the bson module using npm.

**Comment by jskye** [comment]
Im not sure why the tests pass, but i still get this (unless i use the hack fix). 
i still get this problem after updating node and npm and cache clean.
Maybe this needs to be fixed by mongoose people.
I guess i may need to copy the module, to persist the fix for now...
Or otherwise just stick with the pure-js version it switches too.
Is there much difference between the two?

**Comment by mjlescano** [comment]
@jskye There's no much difference. But, It's very likely that this gets fixed updating mongoose (#1224). I think I'll be doing it next week, but, if you find some time before and want to send a PR, that would be awesomee 😊

**Comment by jskye** [comment]
no worries. the fixture problem was just that i overlooked setting the node path before running the command.
`NODE_PATH=. node ./bin/dos-db load tag ./lib/fixtures/tags.json`
